### PR TITLE
Shuffle files around as agreed.

### PR DESCRIPTION
**Pull request by geeknoid**
Also includes minor comment updates and a few
field renames, based on outstanding feedback.

### CODE REVIEW

**Comment by frankbu** [comment]
Where will the proxy/routing config proto definitions go?  I assume something like `api/proxy/v1/config/cfg.proto`? The only issue then is that the current proposal for routing config (subject to change of course) is reusing the Aspect and AspectRule data structures, so we would have a dependency from proxy config to mixer config. Would it make sense to separate out shared types and move them to `api/istio/v1/config/cfg.proto`, for example?

File: mixer/v1/config/aspect/denyChecker.proto [hunk 1]
```diff
@@ -14,7 +14,7 @@
 
 syntax = "proto3";
 
-package istio.config.v1.aspect.denyChecker;
+package istio.mixer.v1.config.aspect;
 
 // Config - User visible configuration of this aspect
 message Config {
```
Comment (douglas-reid) [inline mixer/v1/config/aspect/denyChecker.proto]: did you want to change Config to DenyCheckerConfig or something?

File: mixer/v1/config/cfg.proto [hunk 2]
```diff
@@ -17,11 +17,11 @@ syntax = "proto3";
 import "google/protobuf/struct.proto";
 
 
-package istio.config.v1;
+package istio.nixer.v1.config;
```
Comment (douglas-reid) [inline mixer/v1/config/cfg.proto]: s/nixer/mixer

File: mixer/v1/config/aspect/quota.proto [hunk 3]
```diff
@@ -0,0 +1,21 @@
+// Copyright 2017 Google Inc.
```
Comment (douglas-reid) [inline mixer/v1/config/aspect/quota.proto]: should aspect quota config point to the quota_descriptor in some fashion?  what is that relationship?

File: mixer/v1/config/aspect/quota.proto [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package istio.config.v1.aspect.quota;
+
+// Configures a quota aspect.
+message Quota {
```
Comment (douglas-reid) [inline mixer/v1/config/aspect/quota.proto]: should we call this QuotaConfig?

File: mixer/v1/config/aspect/listChecker.proto [hunk 5]
```diff
@@ -14,12 +14,11 @@
...
 
-package istio.config.v1.aspect.listChecker;
+package istio.mixer.v1.config.aspect;
 
 
-// Config - User visible configuration of this aspect
-// Must be defined as a proto
-message Config {
+// Configures a list checker aspect
+message ListChecker {
```
Comment (douglas-reid) [inline mixer/v1/config/aspect/listChecker.proto]: Should we call this ListCheckerConfig?

File: mixer/v1/config/aspect/quota.proto [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package istio.config.v1.aspect.quota;
```
Comment (douglas-reid) [inline mixer/v1/config/aspect/quota.proto]: istio.mixer.v1.config.aspect (to match others)?

**Comment by geeknoid** [comment]
@frankbu we were thinking of having the following additional directories:

istio.io/api/common/v1/config/...  - common config state
istio.io/api/proxy/v1/...  - proxy API
istio.io/api/proxy/v1/config/... - proxy config.

Does that seem reasonable to you?

File: mixer/v1/config/aspect/quota.proto (hunk 4 above)
Comment (geeknoid) [inline mixer/v1/config/aspect/quota.proto]: Done.

File: mixer/v1/config/aspect/quota.proto (hunk 6 above)
Comment (geeknoid) [inline mixer/v1/config/aspect/quota.proto]: Done

File: mixer/v1/config/aspect/quota.proto (hunk 3 above)
Comment (geeknoid) [inline mixer/v1/config/aspect/quota.proto]: I consider this a placeholder file. I'm working on the quota aspect manager right now so I'll be figuring out the exact content of this file shortly.

File: mixer/v1/config/cfg.proto (hunk 2 above)
Comment (geeknoid) [inline mixer/v1/config/cfg.proto]: Done

File: mixer/v1/config/aspect/denyChecker.proto (hunk 1 above)
Comment (geeknoid) [inline mixer/v1/config/aspect/denyChecker.proto]: Done

File: mixer/v1/config/aspect/listChecker.proto (hunk 5 above)
Comment (geeknoid) [inline mixer/v1/config/aspect/listChecker.proto]: Done

File: mixer/v1/config/aspect/quota.proto (hunk 3 above)
Comment (douglas-reid) [inline mixer/v1/config/aspect/quota.proto]: OK. I'll be interested in the approach, as we will probably want to duplicate it for logger and metrics aspects (i wasn't sure if the aspect config should point to a descriptor or just be the descriptor itself).

**Comment by geeknoid** [comment]
PTAL.

**Comment by geeknoid** [comment]
cfg.proto also includes Adapter config definitions, which aren't common, right?

So I left it in mixer, and we can move pieces out to common as a separate PR.

**Comment by frankbu** [comment]
@geeknoid: yes, I think that mixer/v1/config/cfg.proto needs to be refactored to pull out the common parts, when we finalize the proxy config ... right now it's a mix of common and mixer specific types

**Pull request merged**